packages/zk-eddsa-event-ticket-pcd/test/ZKEdDSAEventTicketPCD.spec.ts

```diff
@@ -0,0 +1,394 @@
+    expect(pcd.claim.nullifierHash).to.be.equal(undefined);
+
+    const verificationRes = await ZKEdDSAEventTicketPCDPackage.verify(pcd1);
+    expect(verificationRes).to.be.true;
+  });
+
+  it("should reveal isConsumed and isRevoked if requested", async function () {
+    const pcdArgs = await toArgs(
+      ticketData1,
+      fieldsToReveal2,
+      true /* withNullifier */,
+      validEventIds1
+    );
+    const pcd = await ZKEdDSAEventTicketPCDPackage.prove(pcdArgs);
+
+    const claim = pcd.claim;
+    expect(claim.partialTicket.isConsumed).to.be.equal(ticketData1.isConsumed);
+    expect(claim.partialTicket.isRevoked).to.be.equal(ticketData1.isRevoked);
+
+    const verificationRes = await ZKEdDSAEventTicketPCDPackage.verify(pcd);
+    expect(verificationRes).to.be.true;
+  });
+
+  it("should prove and verify without any validEventIds", async function () {
+    const pcdArgs = await toArgs(
+      ticketData1,
+      fieldsToReveal1,
+      true /* withNullifier */
+    );
+    const pcd = await ZKEdDSAEventTicketPCDPackage.prove(pcdArgs);
+    expect(pcd.claim.validEventIds).to.be.undefined;
+    const verificationRes = await ZKEdDSAEventTicketPCDPackage.verify(pcd);
+    expect(verificationRes).to.be.true;
+  });
+
+  it("should not prove if identity is not in ticket", async function () {
+    const pcdArgs = await toArgs(
+      ticketData1,
+      fieldsToReveal1,
+      true /* withNullifier */,
+      validEventIds2
+    );
+    pcdArgs.identity = {
+      value: await makeSerializedIdentityPCD(identity2),
+      argumentType: ArgumentTypeName.PCD,
+      pcdType: SemaphoreIdentityPCDTypeName
+    };
+    await assert.rejects(async () => {
+      await ZKEdDSAEventTicketPCDPackage.prove(pcdArgs);
+    });
+  });
+
+  it("should not prove if ticket is not in validEventIds", async function () {
+    const pcdArgs = await toArgs(
+      ticketData1,
+      fieldsToReveal1,
+      true /* withNullifier */,
+      validEventIds2
+    );
+    await assert.rejects(async () => {
+      await ZKEdDSAEventTicketPCDPackage.prove(pcdArgs);
+    });
+  });
+
+  it("should not prove if validEventIds is too large", async function () {
+    const validEventIdsTooLarge = Array<string>(
+      VALID_EVENT_IDS_MAX_LEN + 1
+    ).fill(uuid());
+    const pcdArgs = await toArgs(
+      ticketData1,
+      fieldsToReveal1,
+      true /* withNullifier */,
+      validEventIdsTooLarge
+    );
+    await assert.rejects(async () => {
+      await ZKEdDSAEventTicketPCDPackage.prove(pcdArgs);
+    });
+  });
+
+  it("should not verify a proof with invalid partialTicket fields", async function () {
+    const pcdArgs = await toArgs(
+      ticketData1,
+      fieldsToReveal1,
+      true /* withNullifier */,
+      validEventIds1
+    );
+    const invalidPCD = await ZKEdDSAEventTicketPCDPackage.prove(pcdArgs);
+
+    // set ticketId to be some random uuid
+    invalidPCD.claim.partialTicket.ticketId = uuid();
```

Comment:
I know we don't do this for the other PCDs, but I'm wondering how you feel about testing the fields exhaustively? I don't think it would be much more work than a for loop.

Reply:
I'm not sure if it can be a for loop given the fields don't all have the same type, or the same set of legal values.
There would have to be some manual code for each case to fill in data which is well formed but doesn't match the proof.  It would be a pretty easy sequence of copy/paste to create it, though, and I think it's a reasonable idea.  Let me follow up on that after I send you a new version to review.